File: dev-server.js
// Simple Development Server for EPL News Hub
// Run with: node dev-server.js

const http = require('http');
const fs = require('fs');
const path = require('path');
const url = require('url');

const PORT = process.env.PORT || 8080;
const HOST = '0.0.0.0';

// MIME types
const mimeTypes = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.jpeg': 'image/jpeg',
    '.gif': 'image/gif',
    '.webp': 'image/webp',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon',
    '.woff': 'font/woff',
    '.woff2': 'font/woff2',
    '.ttf': 'font/ttf',
    '.otf': 'font/otf'
};

// Create server
const server = http.createServer((req, res) => {
    // Parse URL
    const parsedUrl = url.parse(req.url);
    let pathname = parsedUrl.pathname;
    
    // Default to index.html for root
    if (pathname === '/') {
        pathname = '/index.html';
    }
    
    // Security: prevent directory traversal
    pathname = pathname.replace(/\.\./g, '');
    
    // Get file path
    const filePath = path.join(__dirname, pathname);
    
    // Check if file exists
    fs.access(filePath, fs.constants.F_OK, (err) => {
        if (err) {
            // File not found
            res.writeHead(404, { 'Content-Type': 'text/html' });
            res.end('<h1>404 - File Not Found</h1>');
            console.log(`404: ${pathname}`);
            return;
        }
        
        // Read and serve file
        fs.readFile(filePath, (err, data) => {
            if (err) {
                res.writeHead(500, { 'Content-Type': 'text/html' });
                res.end('<h1>500 - Internal Server Error</h1>');
                console.error(`Error reading file: ${err}`);
                return;
            }
            
            // Get file extension
            const ext = path.extname(filePath).toLowerCase();
            const contentType = mimeTypes[ext] || 'application/octet-stream';
            
            // Set headers
            res.writeHead(200, {
                'Content-Type': contentType,
                'Access-Control-Allow-Origin': '*',
                'Cache-Control': 'no-cache'
            });
            
            res.end(data);
            console.log(`200: ${pathname}`);
        });
    });
});

// Start server
server.listen(PORT, HOST, () => {
    console.log(`
╔════════════════════════════════════════════════╗
║                                                ║
║     EPL News Hub Development Server            ║
║                                                ║
║     🚀 Server running at:                      ║
║     http://localhost:${PORT}                      ║
║                                                ║
║     📝 Development Features:                   ║
║     - Test mode automatically enabled          ║
║     - Stripe test keys active                  ║
║     - Development banner visible               ║
║                                                ║
║     🔧 Keyboard Shortcuts:                    ║
║     - Ctrl+Shift+D: Toggle dev mode           ║
║     - Ctrl+Shift+T: Show test cards           ║
║                                                ║
║     💳 Test Card: 4242 4242 4242 4242         ║
║                                                ║
║     Press Ctrl+C to stop the server           ║
║                                                ║
╚════════════════════════════════════════════════╝
    `);
    
    // Open browser automatically (optional)
    const platform = process.platform;
    const openCommand = platform === 'darwin' ? 'open' : platform === 'win32' ? 'start' : 'xdg-open';
    require('child_process').exec(`${openCommand} http://localhost:${PORT}/membership.html?dev=true`);
});

File: dev-server.py
#!/usr/bin/env python3
"""
Simple Development Server for EPL News Hub
Run with: python3 dev-server.py
"""

import http.server
import socketserver
import webbrowser
import sys
import os

PORT = 8080
HOST = '0.0.0.0'

class DevHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def end_headers(self):
        # Add CORS headers for development
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Cache-Control', 'no-cache')
        super().end_headers()
    
    def log_message(self, format, *args):
        # Custom log format
        sys.stdout.write(f"[DEV] {self.address_string()} - {format % args}\n")

def main():
    # Change to script directory
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    
    print("""
╔════════════════════════════════════════════════╗
║                                                ║
║     EPL News Hub Development Server            ║
║                                                ║
║     🚀 Server starting at:                     ║
║     http://localhost:{}                      ║
║                                                ║
║     📝 Development Features:                   ║
║     - Test mode automatically enabled          ║
║     - Stripe test keys active                  ║
║     - Development banner visible               ║
║                                                ║
║     🔧 Keyboard Shortcuts (in browser):        ║
║     - Ctrl+Shift+D: Toggle dev mode           ║
║     - Ctrl+Shift+T: Show test cards           ║
║                                                ║
║     💳 Test Card: 4242 4242 4242 4242         ║
║                                                ║
║     Press Ctrl+C to stop the server           ║
║                                                ║
╚════════════════════════════════════════════════╝
    """.format(PORT))
    
    with socketserver.TCPServer((HOST, PORT), DevHTTPRequestHandler) as httpd:
        # Open browser automatically
        url = f"http://localhost:{PORT}/membership.html?dev=true"
        print(f"Opening browser at: {url}")
        webbrowser.open(url)
        
        print(f"\nServer running at http://localhost:{PORT}/")
        print("Press Ctrl+C to stop\n")
        
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\n\nServer stopped.")
            sys.exit(0)

if __name__ == "__main__":
    main()